drivers/postgres/postgres.go

```diff
@@ -208,33 +214,61 @@ func (pg *postgres) Apply(migration *models.Migration, saveVersion bool) (err er
 	ctx, cancel := drivers.GetContext(pg.config.StatementTimeoutInSecs)
 	defer cancel()
 
-	transaction, err := pg.conn.BeginTx(ctx, nil)
-	if err != nil {
-		return &drivers.DatabaseError{
-			OrigErr: err,
-			Driver:  driverName,
-			Message: "error while opening a transaction to the database",
-			Command: "begin_transaction",
+	nonTransactional := strings.HasPrefix(query, "-- "+nonTransactionalPrefix)
+	// We wrap with a transaction only when there is no non-transactional prefix.
+	if !nonTransactional {
+		transaction, err := pg.conn.BeginTx(ctx, nil)
+		if err != nil {
+			return &drivers.DatabaseError{
+				OrigErr: err,
+				Driver:  driverName,
+				Message: "error while opening a transaction to the database",
+				Command: "begin_transaction",
+			}
 		}
-	}
-
-	if err = executeQuery(transaction, query); err != nil {
-		return err
-	}
 
-	if saveVersion {
-		if err = executeQuery(transaction, pg.addMigrationQuery(migration)); err != nil {
+		if err = executeQuery(ctx, transaction, query); err != nil {
 			return err
 		}
-	}
 
-	err = transaction.Commit()
-	if err != nil {
-		return &drivers.DatabaseError{
-			OrigErr: err,
-			Driver:  driverName,
-			Message: "error while committing a transaction to the database",
-			Command: "commit_transaction",
+		if saveVersion {
+			if err = executeQuery(ctx, transaction, pg.addMigrationQuery(migration)); err != nil {
+				return err
+			}
+		}
+
+		err = transaction.Commit()
+		if err != nil {
+			return &drivers.DatabaseError{
+				OrigErr: err,
+				Driver:  driverName,
+				Message: "error while committing a transaction to the database",
+				Command: "commit_transaction",
+			}
+		}
+	} else {
+		_, err := pg.conn.ExecContext(ctx, query)
```

Comment:
Maybe we can avoid the extra indent (suggestion may not compile but I like the UX :p)
```suggestion
		}
		
		return nil
	}

	_, err = pg.conn.ExecContext(ctx, query)
```

Reply:
0/5 :p